packages/yoastseo/src/languageProcessing/languages/ja/customResearches/getWordForms.js

```diff
@@ -0,0 +1,46 @@
+import { get, includes } from "lodash-es";
+import parseSynonyms from "../../../helpers/sanitize/parseSynonyms";
+import createWordForms from "../helpers/internal/createWordForms";
+import getContentWords from "../helpers/getContentWords";
+
+/**
+ * Creates word forms for each word in the given keyphrase.
+ *
+ * @param {string}   keyphrase         The keyphrase to generate word forms for.
+ * @param {Object}   morphologyData    The morphology data to use when generating keyphrase word forms.
+ *
+ * @returns {Array<string[]>} The word forms for each word in the keyphrase.
+ */
+function getKeyphraseForms( keyphrase, morphologyData ) {
+	let keyphraseWords = getContentWords( keyphrase );
+
+	// The keyphrase is in double quotes: use it as an exact match keyphrase.
+	const doubleQuotes = [ "「", "」", "『", "』", "“", "”", "〝", "〞", "〟", "‟", "„", "\"" ];
+	if ( includes( doubleQuotes, keyphrase[ 0 ] ) && includes( doubleQuotes, keyphrase[ keyphrase.length - 1 ] ) ) {
+		keyphrase = keyphrase.substring( 1, keyphrase.length - 1 );
+		keyphraseWords = [ keyphrase ];
```

Comment:
In connection to why exact matches still get forms: I think it's because you don't return here. Even if the if condition applies, you'll move on to `return keyphraseWords.map( word => createWordForms( word, morphologyData ) );
`.

In practice, the only thing you're doing in that case is skipping `getContentWords` and treating the keyphrase as a single word. So for keyphrases with single words it will still create forms as usual, whereas for keyphrases consisting of multiple words it will treat them as one word and create nonsense results because it's skipping the segmentation.